pkg/envfuncs/kind_funcs.go

```diff
@@ -68,16 +76,41 @@ func CreateKindClusterWithConfig(clusterName, image, configFilePath string) env.
 			return ctx, err
 		}
 
-		// stall, wait for pods initializations
-		time.Sleep(7 * time.Second)
-
 		// update envconfig  with kubeconfig
 		cfg.WithKubeconfigFile(kubecfg)
+
+		// stall, wait for pods initializations
+		if err := waitForControlPlane(ctx, cfg.Client()); err != nil {
+			return ctx, err
+		}
+
 		// store entire cluster value in ctx for future access using the cluster name
 		return context.WithValue(ctx, kindContextKey(clusterName), k), nil
 	}
 }
 
+func waitForControlPlane(ctx context.Context, client klient.Client) error {
```

Comment:
This is awesome !!

Reply:
Suggestion: maybe create a pre-built condition wrapper method 

```go
func (c *Condition) ControlPlaneReady(...)
```
Just a suggestion, not necessary, but since the logic is reusable, it may be a nice to have.

Reply:
Right now I have the control-plane readiness defined twice; once in an internal test for the klient/wait package, and another in the envfuncs package.

Creating a re-usable func on the condition type might allow me to remove the duplication -- will think on that. I had avoided it at first out of fear of an import cycle but will revisit.